v3_async_wip/v3_async_wip/mqtt_client.py

```diff
@@ -404,32 +591,12 @@ async def disconnect(self):
         """
         # Wait for permission to alter the connection
         async with self._connection_lock:
-            loop = asyncio.get_running_loop()
-
-            if self._desire_connection:
-                # Record the type of operation in progress
-                self._connection_lock.connection_type = OP_TYPE_DISCONNECT
 
+            # TODO: why can't this be simplified to 'if self._network_loop'
```

Comment:
It isn't immediately apparent, but these SHOULD be identical conditions, yet they aren't in testing. I will not check in until I get to the bottom of it (I suspect it's due to something with the test infrastructure)